network/write.go

```diff
@@ -87,10 +86,17 @@ func (l *write) send(buf []byte, ctx context.Context, retryCount int) sendResult
 		result.networkError = true
 		return result
 	}
+	// Add custom headers from configuration first
+	for key, value := range l.cfg.Headers {
+		httpReq.Header.Set(key, value)
+	}
+
+	// Add/Set required headers, which will override custom headers with the same name
```

Comment:
The add below here stuck out to me as something that looked off but the original implementation uses it.